features/armv8_stack_overflow_guards/README.md

```diff
@@ -0,0 +1,194 @@
+  - Download the latest SDK version for the DA1469x family of devices (10.0.10.x)
+  - **SEGGER's J-Link** tools should be downloaded and installed.
+
+## SDK Modifications
+
+You can apply the modification to the SDK in order to support the ADF feature by using the following command:
+
+```console
+  > cd <SDKROOT>
+  > patch -p1 < <SDK_EXAMPLE_ROOT>/features/features/armv8_stack_overflowguards/sdk.patch 
+```
+
+Alternatively you can apply the modification manually
+
+- **Add MSP Protection in the assembly startup**
+
+  Navigate to startup/DA1469x/GCC/exception_handlers.s and make the following modifications:
+
+  First, add the following lines to the start of the Wakeup_Reset_Handler:
+
+  ```c
+  Wakeup_Reset_Handler:
+
+  /*
+  The MSPLIM register is not retained in sleep, we must restore it on wakeup or from 
+  reset
+  */
+  #if dg_configARMV8_USE_STACK_GUARDS == 1
+      ldr		r0, =__StackLimit
+      add		r0, r0, #dg_configMSP_PADDING
+      msr		MSPLIM, r0
+  #endif
+
+  ```
+
+  Next, add the initialization of the MSPLIM register in the Reset_Handler (After the str r1, [r0]):
+
+  ```c
+
+  Reset_Handler:
+                  ldr     r0, =0x20000004
+                  ldr     r1, =Wakeup_Reset_Handler
+                  str     r1, [r0]
+  #if dg_configARMV8_USE_STACK_GUARDS == 1
+  /*
+  * Intialize the MSPLIM register.  Must be placed here as the msr instruction
+  * requires the device to be in privileged mode. Add padding to the register
+  * so we can leave room for the UsageFault_Handler to place items on the stack
+  */
+          ldr		r0, =__StackLimit
+          add		r0, r0, #dg_configMSP_PADDING
+          msr		MSPLIM, r0
+  #endif
+  ```
+
+  Last, we need to add the UsageFault_Handler in assembly, place this as a seperate handler in the file:
+ ```c
+
+  #if dg_configARMV8_USE_STACK_GUARDS == 1
+#if (dg_configCODE_LOCATION == NON_VOLATILE_IS_FLASH)
+            .section text_retained
+#endif
+             .align  2
+             .thumb
+             .thumb_func
+             .globl  UsageFault_Handler
+             .type   UsageFault_Handler, %function
+ UsageFault_Handler:
+ /*
+	 * We need special handling for the UsageFault_Handler with the use of the MSPLIM register set.  First we check to
+	 * see if the MSPLIM or the PSPLIM registers equal the msp or psp and set the stack pointer mask respectively.
+	 * If the MSPLIM register triggered the usagefault, then we must move the MSPLIM back below the padding to allow for the
+	 * UsageFault_HandlerC to push to the stack if needed.  If the MSPLIM is not moved, then a push to the stack can cause
+	 * nested UsageFaults and circular exceptions
+	 */
+		       ldr		r2,=UsageFault_HandlerC
+		       mrs		r1, msp
+		       mrs		r0, MSPLIM
+		       cmp		r0, r1
+		       beq		UsageFault_with_MSP_Overflow
+		       mrs		r1, psp
+		       mrs		r0, PSPLIM
+		       cmp		r0, r1
+		       beq 		UsageFault_with_PSP_Overflow
+		       mov		r0, #0
+		       bx		r1
```

Comment:
Believe last line above:
```
bx		r1
```
should be:
```
bx		r2
```

You are trying to branch to UsageFault_HandlerC stored in R2. R1 currently holds the psp.